src/libratbag-private.h

```diff
@@ -295,6 +295,7 @@ struct ratbag_profile {
 	unsigned int hz;	/**< report rate in Hz */
 	unsigned int rates[8];	/**< report rates available */
 	size_t nrates;		/**< number of entries in rates */
+	bool rate_dirty;
```

Comment:
Experimental option (just typed out here):

```
struct ratbag_profile {
    [...]
    union {
        unsigned int is_dirty;
        struct {
            bool rate:1;
            bool resolutions:1;
        } dirty;
    };
};
```
This uses a C11 anonymous union and bitfields for the various dirty bits. And because it's a union we only need to check one field to check all of them.
Which would make the code easy to read and easy to check:
```
dev->dirty.rate = true;
if (dev->is_dirty) { ... }
```
as long as `sizeof(is_dirty)` is greater than all bits for the fields, we're good. Could even put a static assert for that in.

But really, I just like having all the dirty bits grouped in a struct to force the code to be readable.

How does that sound?

Reply:
Sure. We can do that in other PR, is that fine?

Reply:
sure, shouldn't block this one.